arches/app/media/js/viewmodels/workflow-step.js

```diff
@@ -219,6 +222,31 @@ define([
             );
         };
 
+        this.setToWorkflowHistory = function(key, value) {
+            const workflowid = self.workflow.id();
+            const workflowHistory = {
+                workflowid,
+                completed: false,
+                stepdata: {
+                    // Django view will patch in this key, keeping existing keys
```

Comment:
Letting the Django view just combine incoming incomplete workflow data avoids requiring each workflow step to their own GET requests to get the current state of the object. I noticed that when doing the GET before a POST back, the data might already be updated on the server by some other POST before the current request can post it back, causing lost data. There were also performance implications and lots of issues awaiting in all the right places while the UI initialized.

(Or, the whole workflow infra could have been refactored to manipulate some shared state object on the frontend, but I figured that wasn't in scope // this was cleaner.)

Reply:
I think this is a nice solution, and glad you added the comment.